### ISSUE TITLE
[Bug]: Chart data not displaying collected points in certain areas

### ISSUE THREAD

**Issue by xkisu** [netdata/netdata#13606]
### Bug description

I was testing some different memory mode configurations on the collector agent (before I saw that was I was trying [wasn't supported](https://github.com/netdata/netdata/pull/9832)) and restarting it a few times (streaming collector to parent), when some charts completely stopped displaying data. I could tell the data was still coming in because some specific charts where still displaying the data in those periods, and the gauges at the top showing the system overview where updating. 

Here's a video showing the problematic area of the chart:
https://user-images.githubusercontent.com/28452108/183015542-4b4a9d39-256e-422a-9f46-a1a163303aaf.mov

After looking at the data a bit, I realized it was actually something weird doing on with the charts displaying the data, and not actual missing data points. If I panned or zoomed in the charts that where missing data, suddenly the missing points would appear correctly. After zooming out they would vanish again, until I zoomed in or selected a region that appeared empty.

I tried changing the chart type on the effected charts to line or area, but it had no effect. Most of the charts where effected, except for a couple as shown in my screenshot. This issue persists after a restart, still showing the same lack of data at the same area of the charts, and not displaying any new data after the effected area.

Checking the dashboard on the agent (via the parent), it's also displaying the data points during that period as missing for most of the charts as well, except the live stats are updating correctly and some of the charts still include the points as seen on the Netdata Cloud dashboard.

The issue stopped once there was enough new points so that the problematic region where the restarts occurred was pushed off the end of the chart, but if I pan back then the charts start vanishing again, see:
https://user-images.githubusercontent.com/28452108/183017418-67a73859-4e26-44fc-919a-f88ed2fd2b12.mov

 

### Expected behavior

Expected the stacked charts to show the collected data points from the agent.

### Steps to reproduce

1. Changed the memory mode on the agent a few times testing settings, config example:
```
[global]
  run as user = netdata

  # default storage size - increase for longer data retention
  page cache size = 32
  dbengine multihost disk space = 256
  memory mode = save # then tried ram, and dbengine to assess the resource usage for the agent

[health]
  enabled = no

[web]
  mode = none
```
2. Restarted the agent `systemctl restart netdata` after changing the memory mode
3. When to check the agent on the dashboard and alarms where showing, stat overview where updating correctly, but most of the charts weren't showing data collected after the restart (even accounting for a delay in the agent restarting).
4. Zooming out or selecting an area on the charts suddenly revealed the missing data, as did panning past the end of the problematic region of points
5. After setting `memory mode = none` back, the problematic display issues stopped for new data but continued for the data while it was set to `ram` and `save` if I panned back.

### Screenshots

Showing the incorrectly displayed data on Netdata Cloud:
![Screen Shot 2022-08-04 at 11 26 49 PM](https://user-images.githubusercontent.com/28452108/183015597-c2a78e11-a644-4bcb-8ce5-1bd3539c6645.png)

Data is also displaying incorrectly on the parent agent's dashboard for the same charts:
![image](https://user-images.githubusercontent.com/28452108/183019184-fe4128dc-508e-4fe3-be78-3a700731f4cf.png)

But as with the cloud, appears when you then zoom into the empty areas:
![image](https://user-images.githubusercontent.com/28452108/183019285-d4fee6a1-7500-40ec-be97-7f85598126c5.png)

### Error Logs

Browser console:
```
[Error] SyntaxError: Can't create duplicate variable: 'contentOriginal'
	write (aptrinsic-widget.js:27:104788)
	value (aptrinsic-widget.js:27:104788)
	value (aptrinsic-widget.js:27:105275)
	u (aptrinsic-widget.js:19:31716)
	o (aptrinsic-widget.js:19:45550)
	a (aptrinsic-widget.js:19:45871)
	s (aptrinsic-widget.js:19:46109)
	w (aptrinsic-widget.js:19:49652)
	_ (aptrinsic-widget.js:19:49389)
	h (aptrinsic-widget.js:19:48517)
	updateContainer (aptrinsic-widget.js:19:52166)
	(anonymous function) (aptrinsic-widget.js:19:63617)
	vn (aptrinsic-widget.js:19:63587)
	(anonymous function) (aptrinsic-widget.js:11:161681)
	t (aptrinsic-widget.js:1:106)
	(anonymous function) (aptrinsic-widget.js:1:454)
	Global Code (aptrinsic-widget.js:1:464)
```

Browser network HAR:
[app.netdata.cloud.har.zip](https://github.com/netdata/netdata-cloud/files/9266007/app.netdata.cloud.har.zip)

### Desktop

OS: macOS Monterey 21.4
Browser: safari
Browser Version: 15.5

### Additional context

The issue appears to be something related to how the charts are displaying the data since the data is there, and shows up depending on how you zoom or pan to it. My guess is it looks like there may there's an issue with how the charts are interpolating the data across points in the sections where the agent was restarted frequently, especially since it seems to persist between the Netdata Cloud dashboard and the local agent dashboard over the same period.

Also, here is a snapshot taken from the parent node containing the problematic datapoints in case they're helpful:
[netdata-20220804-230352-1800.snapshot.zip](https://github.com/netdata/netdata-cloud/files/9266129/netdata-20220804-230352-1800.snapshot.zip)
The problematic display area is from 23:04:45 to 23:29:45

Node ID in the cloud is: `8f9448ca-5efe-48a6-8605-945ea3d2683f`

**Comment by Dim-P** [comment]
@xkisu Did you manage to reproduce this at your home setup? There have been several changes in the agent since you opened this (and even more significant ones coming up), so if you haven't noticed it since, I will go ahead and close it.

**Comment by xkisu** [comment]
> @xkisu Did you manage to reproduce this at your home setup? There have been several changes in the agent since you opened this (and even more significant ones coming up), so if you haven't noticed it since, I will go ahead and close it.

Sorry for the delay - yeah I've managed to reproduce it a few times in the last 2-3 weeks, both on my homelab and on another client's servers using the latest nightly agent builds. Every time it's happened after something completely different though so I've not been able to reliably reproduce it or capture a snapshot.

Sometimes it happens from an agent restart, sometimes from a config change, and a couple times I've had it where there's a gap in data from a drop in network connectivity and it starts happening in the gap area even though there are points there that where collected during the outage (if I zoom in they appear i the blank space like in the initial case I reported).
